src/components/Table/GaiaChainTable.tsx

```diff
@@ -0,0 +1,253 @@
+  rowHeight?: number;
+}
+
+class MuiVirtualizedTable extends React.PureComponent<MuiVirtualizedTableProps> {
+  static defaultProps = {
+    headerHeight: 48,
+    rowHeight: 48,
+  };
+
+  getRowClassName = ({ index }: Row) => {
+    const { classes, onRowClick } = this.props;
+
+    return clsx(classes.tableRow, classes.flexContainer, {
+      [classes.tableRowHover]: index !== -1 && onRowClick != null,
+    });
+  };
+
+  cellRenderer: TableCellRenderer = ({ cellData, columnIndex }: any) => {
+    const { columns, classes, rowHeight, onRowClick } = this.props;
+    return (
+      <TableCell
+        component="div"
+        className={clsx(classes.tableCell, classes.flexContainer, {
+          [classes.noClick]: onRowClick == null,
+        })}
+        variant="body"
+        style={{ height: rowHeight }}
+        align={(columnIndex != null && columns[columnIndex].numeric) || false ? 'right' : 'left'}
+      >
+        {cellData}
+      </TableCell>
+    );
+  };
+
+  headerRenderer = ({ label, columnIndex }: TableHeaderProps & { columnIndex: number }) => {
+    const { headerHeight, columns, classes } = this.props;
+
+    return (
+      <TableCell
+        component="div"
+        className={clsx(classes.head, classes.tableCell, classes.flexContainer, classes.noClick)}
+        variant="head"
+        style={{ height: headerHeight }}
+        align={columns[columnIndex].numeric || false ? 'right' : 'left'}
+      >
+        <span>{label}</span>
+      </TableCell>
+    );
+  };
+
+  render() {
+    const { classes, columns, rowHeight, headerHeight, ...tableProps } = this.props;
+    return (
+      <AutoSizer>
+        {({ height, width }: any) => (
+          <Table
+            height={height}
+            width={width}
+            rowHeight={rowHeight!}
+            gridStyle={{
+              direction: 'inherit',
+            }}
+            headerHeight={headerHeight!}
+            className={classes.table}
+            {...tableProps}
+            rowClassName={this.getRowClassName}
+          >
+            {columns.map(({ dataKey, ...other }, index) => {
+              return (
+                <Column
+                  key={dataKey}
+                  headerRenderer={(headerProps: any) =>
+                    this.headerRenderer({
+                      ...headerProps,
+                      columnIndex: index,
+                    })
+                  }
+                  className={classes.flexContainer}
+                  cellRenderer={this.cellRenderer}
+                  dataKey={dataKey}
+                  {...other}
+                />
+              );
+            })}
+          </Table>
+        )}
+      </AutoSizer>
+    );
+  }
+}
+
+const VirtualizedTable = withStyles(styles)(MuiVirtualizedTable);
+
+// ---
+
+interface Data {
+  channels: string;
+  packetCommitment: number;
+  connections: string;
+  port: number;
+  id: number;
+  data: number;
+}
+type Sample = [string, string, number, number, number];
```

Comment:
i think it would make sense to take all this creating dummy data related functionality/variables into a separate utils file and import it from there, we can use it later for testing